src/wallet/rpcnames.cpp

```diff
@@ -515,6 +591,70 @@ name_firstupdate ()
                             "this name is already active");
     }
 
+  uint256 prevTxid = uint256::ZERO; // if it can't find a txid, force an error
+  if (fixedTxid)
+    {
+      prevTxid = ParseHashV (request.params[2], "txid");
+    }
+  else
+    {
+      // Code slightly duplicates name_scan, but not enough to be able to refactor.
+      /* Make sure the results are valid at least up to the most recent block
+         the user could have gotten from another RPC command prior to now.  */
+      pwallet->BlockUntilSyncedToCurrentChain ();
+
+      LOCK (cs_main);
+
+      for (const auto& item : pwallet->mapWallet)
+        {
+          const CWalletTx& tx = item.second;
+          if (!tx.tx->IsNamecoin ())
+            continue;
+
+          CScript output;
+          CNameScript nameOp;
+          bool found = false;
+          for (CTxOut curOutput : tx.tx->vout)
+            {
+              CScript curScript = curOutput.scriptPubKey;
+              const CNameScript cur(curScript);
+              if (!cur.isNameOp ())
+                continue;
+              if (cur.getNameOp () != OP_NAME_NEW)
+                continue;
+              if (found) {
+                LogPrintf ("ERROR: wallet contains tx with multiple"
+                           " name outputs");
+                continue;
+              }
+              nameOp = cur;
+              found = true;
+              output = curScript;
+            }
+
+          if (!found)
+            continue; // no name outputs found
+
+          if (!fixedRand)
+            {
+              bool got = getNameSalt (pwallet, name, output, rand);
```

Comment:
Since you removed the `assert` here completely, there is no need for the temporary variable anymore.

Reply:
Sure